staging/src/k8s.io/client-go/examples/create-update-delete-deployment/main.go

```diff
@@ -138,11 +139,17 @@ func main() {
 			panic(fmt.Errorf("Failed to get latest version of Deployment: %v", getErr))
 		}
 
-		result.Spec.RollbackTo = &appsv1beta1.RollbackConfig{
+		rollBackTo := &appsv1beta1.RollbackConfig{
 			Revision: 0, // can be specific revision number, or 0 for last revision
 		}
-		_, updateErr := deploymentsClient.Update(result)
-		return updateErr
+
+		deploymentRollBack := &appsv1beta1.DeploymentRollBack{
+			Name:       "demo-deployment",
+			RollBackTo: rollBackTo,
+		}
+
+		err := deploymentsClient.Rollback(deploymentRollBack)
```

Comment:
We deprecated `.rollbackTo` field and `/rollback` endpoint altogether. They're not supported in apps/v1.Deployments. Because `deploymentsClient` points to apps/v1.Deployments, I don't expect this to work.

Reply:
@janetkuo @sttts @nikhita @jekohk How deployment rollback happens then for v1? I couldn't find a reference [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#-strong-misc-operations-strong-) like the one mentioned for v1beta1 extension [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.9/#rollback-811) , further [rollback.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/rollback.go) seems to use [extv1beta1](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/rollback.go#L118) , let me know your thoughts please.

Reply:
Because of the removal of /rollback endpoint in apps/v1.Deployments, the example and kubectl, if switched to apps/v1.Deployments, need to do the rollback logic themselves. That includes:

1. List all ReplicaSets the Deployment owns
2. Find the ReplicaSet of a specific revision 
3. Copy that ReplicaSet's template back to the Deployment's template 

The rollback logic currently lives in Deployment controller code, which still uses extensions/v1beta1 Deployment client:
https://github.com/kubernetes/kubernetes/blob/ecc5eb67d965295db95ba2df5f3d3ff43a258a05/pkg/controller/deployment/rollback.go#L30-L69

Reply:
@janetkuo Thanks, will update the PR accordingly.